remoting/subsystem/src/main/java/org/jboss/as/remoting/ConnectorUtils.java

```diff
@@ -156,16 +156,19 @@ private static void addSasl(OperationContext context, ModelNode sasl, OptionMap.
     private static void addOptions(OperationContext context, ModelNode properties, OptionMap.Builder builder) throws OperationFailedException {
         final ClassLoader loader = WildFlySecurityManager.getClassLoaderPrivileged(ConnectorUtils.class);
         for (Property property : properties.asPropertyList()) {
-            String name = property.getName();
-            if (!name.contains(".")) {
-                name = "org.xnio.Options." + name;
-            }
-            final Option option = Option.fromString(name, loader);
+            final Option option = getAndValidateOption(loader, property.getName());
             String value = PropertyResource.VALUE.resolveModelAttribute(context, property.getValue()).asString();
             builder.set(option, option.parseValue(value, loader));
         }
     }
 
+    static Option<?> getAndValidateOption(ClassLoader loader, String name) {
+        if (!name.contains(".")) {
+            name = "org.xnio.Options." + name;
+        }
+        return Option.fromString(name, loader);
```

Comment:
This should catch any IAE and throw an OperationFailedException. An IAE will propagate up and be handled as a server fault, and this is not a server fault. OFE is for user mistakes.

The existing code (i.e. L163) of course had this same problem. :)

Reply:
@bstansberry Fixed